### ISSUE TITLE
Unique path_prefix

### ISSUE THREAD

**Issue by phy1729** [zsh-users/zsh-syntax-highlighting#514]
Possibly add a new style `path_prefix_unique` when the path prefix only matches a single file for a hint to hit tab.

**Comment by danielshahaf** [comment]
Reminds of of #148, which was a similar idea for command words.

**Comment by phy1729** [comment]
Should `path_prefix_unique_pathseparator` fall back to `path_prefix_unique` or `path_prefix_pathseparator` or something else?

**Comment by danielshahaf** [comment]
I'd have voted for the latter, since that would maintain the distinction between a terminal separator and the characters before it, and those characters would be highlighted as `path_prefix_unique` anyway which would serve to convey the uniqueness...

...but I don't highlight path separators differently to other characters in pathnames.  Let's ask the authors of the pathseparators PR (#136 #260).

@blaenk @m0vie WDYT regarding @phy1729's question?

**Comment by blaenk** [comment]
> I'd have voted for the latter, since that would maintain the distinction between a terminal separator and the characters before it, and those characters would be highlighted as path_prefix_unique anyway which would serve to convey the uniqueness...

Thanks for the ping. IIUC, I do agree with this. It'd feel weird to have the path separators suddenly change/remove highlighting, reminiscent of the flickering that (used to?) happen.

**Comment by phy1729** [comment]
> those characters would be highlighted as path_prefix_unique anyway which would serve to convey the uniqueness...

Right, indeed the latter makes more sense. I'll open a PR after #512 since they'll conflict.

**Comment by m0vie** [comment]
> I'd have voted for the latter, since that would maintain the distinction between a terminal separator and the characters before it, and those characters would be highlighted as path_prefix_unique anyway which would serve to convey the uniqueness...

I also agree with this.

**Comment by Sarke** [comment]
I think this is a great suggestion.  How can this be implemented?

**Comment by danielshahaf** [comment]
There's a `(( $#tmp > 0 )) && REPLY=path_prefix` on line 978 in current HEAD (I can't permalink, sorry). Check if `! (( $+tmp[2] ))`. We may be able to optimize this further with glob qualifiers: for example, `integer foo; f() { (( ++foo )) }; : *(N+f)`.

**Comment by danielshahaf** [comment]
There's also `*(Y2)` but it's not available in all supported zsh versions.

Finally, I wonder if perchance there already exists a zsh plugin that provides this functionality.
